### ISSUE TITLE
Promise-manipulation shorthands have underdefined behavior

### ISSUE THREAD

**Issue by bzbarsky** [w3ctag/promises-guide#52]
https://www.w3.org/2001/tag/doc/promises-guide#shorthand-manipulating says:

> If the algorithm using these shorthands is running in parallel, the shorthands queue a task on p’s relevant settings object’s responsible event loop to call the stored function.

But to queue a task you need to know the event loop _and_ the task queue to use.  Which task queue is used here?

**Comment by annevk** [comment]
I think I mentioned this somewhere that this guide basically cannot be used directly by specifications.

@tobie or someone else should take the time to integrate it into IDL as only IDL-derived specifications will be in need of this kind of language.

**Comment by domenic** [comment]
Another issue is that people often want to use them on Web IDL objects, which are realm-agnostic. But we need to call the ES-spec algorithms with JS objects, which are not. So we need a specific global in which to do the conversion from IDL object -> JS object conversion.

I was going to say that this should be the global of the promise by default, to save people typing. But then I remembered that promises, as ES spec objects, don't have associated globals. So that's fun.

**Comment by domenic** [comment]
Good point, I got confused because my first example was a sequence of strings. I'll work to rephrase https://github.com/whatwg/html/pull/2774.

**Comment by bzbarsky** [comment]
Right, sequences have the same problem as dictionaries...

**Comment by domenic** [comment]
Fixed by https://github.com/heycam/webidl/issues/490!
